Framework/Algorithms/src/CropToComponent.cpp

```diff
@@ -99,12 +81,13 @@ void CropToComponent::exec() {
       getProperty("InputWorkspace");
 
   // Get all detectors
-  std::vector<Mantid::Geometry::IDetector_const_sptr> detectors;
-  getDetectors(inputWorkspace, componentNames, detectors);
+  const auto &detectorIndices =
```

Comment:
Just out of interest. The exposure of detector indices is what you eventually want to remove, right?

Reply:
Not really. Instrument-2.0 works preferably with detector indices, and can thus avoid map lookups. In general I think detector IDs should only be used for UI or in files.

Reply:
I thought that detector indices were an implementation detail with no real meaning, except for internal book-keeping. It seems odd that this needs to be handled on an algorithm level.

Reply:
Detector indices are to detector IDs a bit like workspace indices are to spectrum numbers. To be more precise, detector index is even more well defined than workspace index: A workspace index for a spectrum can change for many reasons, such as loading only some spectra, extracting spectra, or reordering spectrum. A detector index is always the same -- it is simply the index into a sorted vector of detector IDs. It only changes if the instrument changes by adding detectors.

You are right though that we do want to keep them "internal" and not expose them to users via algorithms since adding a fourth index would be too confusing. Any access to the Instrument-2.0 API via `DetectorInfo` is using detector indices though.

Reply:
It is a good point that the workspace index itself is only an implementation detail